File: leetcode/rust/snake-in-matrix.rs
impl Solution {
    pub fn final_position_of_snake(n: i32, commands: Vec<String>) -> i32 {
        let mut result = 0;

        for command in commands {
            match command.as_str() {
                "UP" => result -= n,
                "DOWN" => result += n,
                "LEFT" => result -= 1,
                "RIGHT" => result += 1,
                _ => (),
            };
        }

        result
    }
}

File: leetcode/python/snake-in-matrix.py
class Solution:
    def finalPositionOfSnake(self, n: int, commands: list[str]) -> int:
        command_dict = {
            "UP": -n,
            "RIGHT": 1,
            "DOWN": n,
            "lEFT": -1
        }

        return sum([command_dict[command] for command in commands])